MediaBrowser.Common/Extensions/DictionaryExtensions.cs

```diff
@@ -0,0 +1,69 @@
+using System.Collections.Generic;
+
+namespace MediaBrowser.Common.Extensions
+{
+    /// <summary>
+    /// Static extensions for the <see cref="IReadOnlyDictionary{TKey,TValue}"/> interface.
+    /// </summary>
+    public static class DictionaryExtensions
+    {
+        /// <summary>
+        /// Gets a string from a string dictionary, checking all keys sequentially,
+        /// stopping at the first key that returns a result that's neither null nor blank.
+        /// </summary>
+        /// <param name="dictionary">The dictionary.</param>
+        /// <param name="key1">The first checked key.</param>
+        /// <returns>System.String.</returns>
+        public static string? GetFirstNonNullNorBlankDictionaryValue(this IReadOnlyDictionary<string, string>? dictionary, string key1)
+        {
+            return dictionary.GetFirstNonNullNorBlankDictionaryValue(key1, string.Empty, string.Empty);
+        }
+
+        /// <summary>
+        /// Gets a string from a string dictionary, checking all keys sequentially,
+        /// stopping at the first key that returns a result that's neither null nor blank.
+        /// </summary>
+        /// <param name="dictionary">The dictionary.</param>
+        /// <param name="key1">The first checked key.</param>
+        /// <param name="key2">The second checked key.</param>
+        /// <returns>System.String.</returns>
+        public static string? GetFirstNonNullNorBlankDictionaryValue(this IReadOnlyDictionary<string, string>? dictionary, string key1, string key2)
+        {
+            return dictionary.GetFirstNonNullNorBlankDictionaryValue(key1, key2, string.Empty);
+        }
+
+        /// <summary>
+        /// Gets a string from a string dictionary, checking all keys sequentially,
+        /// stopping at the first key that returns a result that's neither null nor blank.
+        /// </summary>
+        /// <param name="dictionary">The dictionary.</param>
+        /// <param name="key1">The first checked key.</param>
+        /// <param name="key2">The second checked key.</param>
+        /// <param name="key3">The third checked key.</param>
+        /// <returns>System.String.</returns>
+        public static string? GetFirstNonNullNorBlankDictionaryValue(this IReadOnlyDictionary<string, string>? dictionary, string key1, string key2, string key3)
```

Comment:
Been thinking about the name. Maybe `GetFirstNotNullNorEmptyValue`?

Reply:
Right, the `Dictionary` can be removed. I'd keep the `Blank` though or, better, change it to `WhiteSpace` to make it obvious that it doesn't return whitespace.